include/aws_iot_mqtt_client_interface.h

```diff
@@ -58,154 +58,242 @@ extern "C" {
+/* @[declare_mqtt_free] */
 
 /**
- * @brief MQTT Connection Function
+ * @brief Establish a connection with an MQTT server.
  *
- * Called to establish an MQTT connection with the AWS IoT Service
+ * This function should be called once and after @ref mqtt_function_init. It sends
+ * the MQTT CONNECT packet to the server, which establishes an MQTT session. Once
+ * the session is no longer needed, it can be closed with @ref mqtt_function_disconnect.
  *
- * @param pClient Reference to the IoT Client
- * @param pConnectParams Pointer to MQTT connection parameters
+ * @param[in] pClient MQTT client context
+ * @param[in] pConnectParams MQTT connection parameters
  *
- * @return An IoT Error Type defining successful/failed connection
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_connect] */
 IoT_Error_t aws_iot_mqtt_connect(AWS_IoT_Client *pClient, IoT_Client_Connect_Params *pConnectParams);
+/* @[declare_mqtt_connect] */
 
 /**
- * @brief Publish an MQTT message on a topic
+ * @brief Publish an MQTT message to a topic.
  *
- * Called to publish an MQTT message on a topic.
- * @note Call is blocking.  In the case of a QoS 0 message the function returns
- * after the message was successfully passed to the TLS layer.  In the case of QoS 1
- * the function returns after the receipt of the PUBACK control packet.
+ * This function sends an MQTT message to the server. The server will then
+ * forward this message to any clients with subscriptions to topic filters
+ * that match the message's topic.
  *
- * @param pClient Reference to the IoT Client
- * @param pTopicName Topic Name to publish to
+ * For a QoS 0 message, this function returns after the message is successfully
+ * passed to the TLS layer. For a QoS 1 message, this function returns after the
+ * receipt of the PUBACK for the transmitted message.
+ *
+ * @param pClient MQTT client context
+ * @param pTopicName Topic name to publish to
  * @param topicNameLen Length of the topic name
- * @param pParams Pointer to Publish Message parameters
+ * @param pParams Publish message parameters
  *
- * @return An IoT Error Type defining successful/failed publish
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_publish] */
 IoT_Error_t aws_iot_mqtt_publish(AWS_IoT_Client *pClient, const char *pTopicName, uint16_t topicNameLen,
 								 IoT_Publish_Message_Params *pParams);
+/* @[declare_mqtt_publish] */
 
 /**
  * @brief Subscribe to an MQTT topic.
  *
- * Called to send a subscribe message to the broker requesting a subscription
- * to an MQTT topic.
- * @note Call is blocking.  The call returns after the receipt of the SUBACK control packet.
- * @warning pTopicName and pApplicationHandlerData need to be static in memory.
+ * This function sends an MQTT subscribe packet to the server. It registers
+ * a subscription that will cause the provided callback function to be invoked
+ * when the server sends a matching message to the client.
```

Comment:
"..._matching message to the client._" -> "..when the server sends message _on the matching topic_ to the client"

Reply:
fixed